app/assets/javascripts/admin_legacy/views/promotional_features/_form.js

```diff
@@ -0,0 +1,48 @@
+(function () {
+  'use strict'
+  window.GOVUK = window.GOVUK || {}
+
+  var promotionalFeaturesForm = {
+    init: function init (options) {
+      this.form = document.querySelector(options.selector)
+      this.toggleImageAndYoutubeFieldsVisibility()
+      this.addRadioButtonEventListeners()
+    },
+
+    toggleImageAndYoutubeFieldsVisibility: function toggleImageAndYoutubeFieldsVisibility () {
+      var form = this.form
+      var imageRadioButtonChecked = form.querySelector('#promotional_feature_promotional_feature_items_attributes_0_image_or_youtube_video_url_image').checked
+      var youtubeRadioButtonChecked = form.querySelector('#promotional_feature_promotional_feature_items_attributes_0_image_or_youtube_video_url_youtube_video_url').checked
+
+      if (!imageRadioButtonChecked) {
+        var imageFields = form.querySelector('.image-fields')
+        imageFields.style.display = 'none'
+      }
+
+      if (!youtubeRadioButtonChecked) {
+        var youtubeFields = form.querySelector('.youtube-video-url-fields')
+        youtubeFields.style.display = 'none'
+      }
+    },
+
+    addRadioButtonEventListeners: function addRadioButtonEventListeners () {
+      var form = this.form
+      var imageRadioButton = form.querySelector('#promotional_feature_promotional_feature_items_attributes_0_image_or_youtube_video_url_image')
+      var youtubeRadioButton = form.querySelector('#promotional_feature_promotional_feature_items_attributes_0_image_or_youtube_video_url_youtube_video_url')
+      var imageFields = form.querySelector('.image-fields')
+      var youtubeFields = form.querySelector('.youtube-video-url-fields')
+
+      imageRadioButton.addEventListener('click', function () {
+        imageFields.style.display = 'block'
```

Comment:
🙋🏻‍♂️ just to chime in here – I think an even nicer approach would be to set the `hidden` attribute of the element, rather than just setting the CSS styles.

e.g.

```
imageFields.hidden = true
```

and

```
imageFields.hidden = false
```

I think the `hidden` attribute has more semantic meaning – and it also means you don't have to do obscure things like setting `style.display = ''` and confusing future developers who might not realise that `''` empty strings reset CSS rules.

Reply:
thanks have sorted it